doc/man/neoxad.1
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.47.11.
.TH NEOXAD "1" December 2021" "neoxad v0.17.0.0" "User Commands"
.SH NAME
neoxad \- manual page for neoxad v0.17.0.0
.SH DESCRIPTION
Neoxa Core Daemon version v0.17.0.0
.SS "Usage:"
.TP
neoxad [options]
Start Neoxa Core Daemon
.SH OPTIONS
.HP
\-?
.IP
Print this help message and exit
.HP
\fB\-alertnotify=\fR<cmd>
.IP
Execute command when a relevant alert is received or we see a really
long fork (%s in cmd is replaced by message)
.HP
\fB\-assumevalid=\fR<hex>
.IP
If this block is in the chain assume that it and its ancestors are valid
and potentially skip their script verification (0 to verify all,
default:
00000000000000105cfae44a995332d8ec256850ea33a1f7b700474e3dad82bc,
testnet:
0000009303aeadf8cf3812f5c869691dbd4cb118ad20e9bf553be434bafe6a52)
.HP
\fB\-blocknotify=\fR<cmd>
.IP
Execute command when the best block changes (%s in cmd is replaced by
block hash)
.HP
\fB\-blockreconstructionextratxn=\fR<n>
.IP
Extra transactions to keep in memory for compact block reconstructions
(default: 100)
.HP
\fB\-blocksdir=\fR<dir>
.IP
Specify blocks directory (default: <datadir>/blocks)
.HP
\fB\-conf=\fR<file>
.IP
Specify configuration file. Relative paths will be prefixed by datadir
location. (default: neoxa.conf)
.HP
\fB\-daemon\fR
.IP
Run in the background as a daemon and accept commands
.HP
\fB\-datadir=\fR<dir>
.IP
Specify data directory
.HP
\fB\-dbcache=\fR<n>
.IP
Set database cache size in megabytes (4 to 16384, default: 300)
.HP
\fB\-debuglogfile=\fR<file>
.IP
Specify location of debug log file. Relative paths will be prefixed by a
net\-specific datadir location. (0 to disable; default: debug.log)
.HP
\fB\-loadblock=\fR<file>
.IP
Imports blocks from external blk000??.dat file on startup
.HP
\fB\-maxmempool=\fR<n>
.IP
Keep the transaction memory pool below <n> megabytes (default: 300)
.HP
\fB\-maxorphantxsize=\fR<n>
.IP
Maximum total size of all orphan transactions in megabytes (default: 10)
.HP
\fB\-maxrecsigsage=\fR<n>
.IP
Number of seconds to keep LLMQ recovery sigs (default: 604800)
.HP
\fB\-mempoolexpiry=\fR<n>
.IP
Do not keep transactions in the mempool longer than <n> hours (default:
336)
.HP
\fB\-par=\fR<n>
.IP
Set the number of script verification threads (\fB\-4\fR to 16, 0 = auto, <0 =
leave that many cores free, default: 0)
.HP
\fB\-persistmempool\fR
.IP
Whether to save the mempool on shutdown and load on restart (default: 1)
.HP
\fB\-pid=\fR<file>
.IP
Specify pid file. Relative paths will be prefixed by a net\-specific
datadir location. (default: neoxad.pid)
.HP
\fB\-prune=\fR<n>
.IP
Reduce storage requirements by enabling pruning (deleting) of old
blocks. This allows the pruneblockchain RPC to be called to
delete specific blocks, and enables automatic pruning of old
blocks if a target size in MiB is provided. This mode is
incompatible with \fB\-txindex\fR, \fB\-rescan\fR and \fB\-disablegovernance\fR=\fI\,false\/\fR.
Warning: Reverting this setting requires re\-downloading the
entire blockchain. (default: 0 = disable pruning blocks, 1 =
allow manual pruning via RPC, >945 = automatically prune block
files to stay under the specified target size in MiB)
.HP
\fB\-syncmempool\fR
.IP
Sync mempool from other nodes on start (default: 1)
.HP
\fB\-sysperms\fR
.IP
Create new files with system default permissions, instead of umask 077
(only effective with disabled wallet functionality)
.HP
\fB\-version\fR
.IP
Print version and exit
.PP
Connection options:
.HP
\fB\-addnode=\fR<ip>
.IP
Add a node to connect to and attempt to keep the connection open (see
the `addnode` RPC command help for more info). This option can be
specified multiple times to add multiple nodes.
.HP
\fB\-allowprivatenet\fR
.IP
Allow RFC1918 addresses to be relayed and connected to (default: 0)
.HP
\fB\-banscore=\fR<n>
.IP
Threshold for disconnecting misbehaving peers (default: 100)
.HP
\fB\-bantime=\fR<n>
.IP
Number of seconds to keep misbehaving peers from reconnecting (default:
86400)
.HP
\fB\-bind=\fR<addr>
.IP
Bind to given address and always listen on it. Use [host]:port notation
for IPv6
.HP
\fB\-connect=\fR<ip>
.IP
Connect only to the specified node; \fB\-connect\fR=\fI\,0\/\fR disables automatic
connections (the rules for this peer are the same as for
\fB\-addnode\fR). This option can be specified multiple times to connect
to multiple nodes.
.HP
\fB\-discover\fR
.IP
Discover own IP addresses (default: 1 when listening and no \fB\-externalip\fR
or \fB\-proxy\fR)
.HP
\fB\-dns\fR
.IP
Allow DNS lookups for \fB\-addnode\fR, \fB\-seednode\fR and \fB\-connect\fR (default: 1)
.HP
\fB\-dnsseed\fR
.IP
Query for peer addresses via DNS lookup, if low on addresses (default: 1
unless \fB\-connect\fR used)
.HP
\fB\-enablebip61\fR
.IP
Send reject messages per BIP61 (default: 1)
.HP
\fB\-externalip=\fR<ip>
.IP
Specify your own public address
.HP
\fB\-forcednsseed\fR
.IP
Always query for peer addresses via DNS lookup (default: 0)
.HP
\fB\-listen\fR
.IP
Accept connections from outside (default: 1 if no \fB\-proxy\fR or \fB\-connect\fR)
.HP
\fB\-listenonion\fR
.IP
Automatically create Tor hidden service (default: 1)
.HP
\fB\-maxconnections=\fR<n>
.IP
Maintain at most <n> connections to peers (temporary service connections
excluded) (default: 125)
.HP
\fB\-maxreceivebuffer=\fR<n>
.IP
Maximum per\-connection receive buffer, <n>*1000 bytes (default: 5000)
.HP
\fB\-maxsendbuffer=\fR<n>
.IP
Maximum per\-connection send buffer, <n>*1000 bytes (default: 1000)
.HP
\fB\-maxtimeadjustment\fR
.IP
Maximum allowed median peer time offset adjustment. Local perspective of
time may be influenced by peers forward or backward by this
amount. (default: 4200 seconds)
.HP
\fB\-maxuploadtarget=\fR<n>
.IP
Tries to keep outbound traffic under the given target (in MiB per 24h),
0 = no limit (default: 0)
.HP
\fB\-onion=\fR<ip:port>
.IP
Use separate SOCKS5 proxy to reach peers via Tor hidden services
(default: \fB\-proxy\fR)
.HP
\fB\-onlynet=\fR<net>
.IP
Make outgoing connections only through network <net> (ipv4, ipv6 or
onion). Incoming connections are not affected by this option.
This option can be specified multiple times to allow multiple
networks.
.HP
\fB\-peerbloomfilters\fR
.IP
Support filtering of blocks and transaction with bloom filters (default:
1)
.HP
\fB\-peertimeout=\fR<n>
.IP
Specify p2p connection timeout in seconds. This option determines the
amount of time a peer may be inactive before the connection to it
is dropped. (minimum: 1, default: 60)
.HP
\fB\-permitbaremultisig\fR
.IP
Relay non\-P2SH multisig (default: 1)
.HP
\fB\-port=\fR<port>
.IP
Listen for connections on <port> (default: 9999 or testnet: 19999)
.HP
\fB\-proxy=\fR<ip:port>
.IP
Connect through SOCKS5 proxy
.HP
\fB\-proxyrandomize\fR
.IP
Randomize credentials for every proxy connection. This enables Tor
stream isolation (default: 1)
.HP
\fB\-seednode=\fR<ip>
.IP
Connect to a node to retrieve peer addresses, and disconnect. This
option can be specified multiple times to connect to multiple
nodes.
.HP
\fB\-socketevents=\fR<mode>
.IP
Socket events mode, which must be one of 'select', 'poll', 'epoll' or
\&'kqueue', depending on your system (default: Linux \- 'epoll',
FreeBSD/Apple \- 'kqueue', Windows \- 'select')
.HP
\fB\-socketevents=\fR<mode>
.IP
Socket events mode, which must be one of: 'select' (default: select)
.HP
\fB\-timeout=\fR<n>
.IP
Specify connection timeout in milliseconds (minimum: 1, default: 5000)
.HP
\fB\-torcontrol=\fR<ip>:<port>
.IP
Tor control port to use if onion listening enabled (default:
127.0.0.1:9051)
.HP
\fB\-torpassword=\fR<pass>
.IP
Tor control port password (default: empty)
.HP
\fB\-upnp\fR
.IP
Use UPnP to map the listening port (default: 0)
.HP
\fB\-whitebind=\fR<addr>
.IP
Bind to given address and whitelist peers connecting to it. Use
[host]:port notation for IPv6
.HP
\fB\-whitelist=\fR<IP address or network>
.IP
Whitelist peers connecting from the given IP address (e.g. 1.2.3.4) or
CIDR notated network (e.g. 1.2.3.0/24). Can be specified multiple
times. Whitelisted peers cannot be DoS banned and their
transactions are always relayed, even if they are already in the
mempool, useful e.g. for a gateway
.PP
Indexing options:
.HP
\fB\-addressindex\fR
.IP
Maintain a full address index, used to query for the balance, txids and
unspent outputs for addresses (default: 0)
.HP
\fB\-reindex\fR
.IP
Rebuild chain state and block index from the blk*.dat files on disk
.HP
\fB\-reindex\-chainstate\fR
.IP
Rebuild chain state from the currently indexed blocks
.HP
\fB\-spentindex\fR
.IP
Maintain a full spent index, used to query the spending txid and input
index for an outpoint (default: 0)
.HP
\fB\-timestampindex\fR
.IP
Maintain a timestamp index for block hashes, used to query blocks hashes
by a range of timestamps (default: 0)
.HP
\fB\-txindex\fR
.IP
Maintain a full transaction index, used by the getrawtransaction rpc
call (default: 1)
.PP
Masternode options:
.HP
\fB\-llmq\-data\-recovery=\fR<n>
.IP
Enable automated quorum data recovery (default: 1)
.HP
\fB\-llmq\-qvvec\-sync=\fR<quorum_name>:<mode>
.IP
Defines from which LLMQ type the masternode should sync quorum
verification vectors. Can be used multiple times with different
LLMQ types. <mode>: 0 (sync always from all quorums of the type
defined by <quorum_name>), 1 (sync from all quorums of the type
defined by <quorum_name> if a member of any of the quorums)
.HP
\fB\-masternodeblsprivkey=\fR<hex>
.IP
Set the masternode BLS private key and enable the client to act as a
masternode
.HP
\fB\-platform\-user=\fR<user>
.IP
Set the username for the "platform user", a restricted user intended to
be used by Dash Platform, to the specified username.
.PP
Statsd options:
.HP
\fB\-statsenabled\fR
.IP
Publish internal stats to statsd (default: 0)
.HP
\fB\-statshost=\fR<ip>
.IP
Specify statsd host (default: 127.0.0.1)
.HP
\fB\-statshostname=\fR<ip>
.IP
Specify statsd host name (default: )
.HP
\fB\-statsns=\fR<ns>
.IP
Specify additional namespace prefix (default: )
.HP
\fB\-statsperiod=\fR<seconds>
.IP
Specify the number of seconds between periodic measurements (default:
60)
.HP
\fB\-statsport=\fR<port>
.IP
Specify statsd port (default: 8125)
.PP
Wallet options:
.HP
\fB\-createwalletbackups=\fR<n>
.IP
Number of automatic wallet backups (default: 10)
.HP
\fB\-disablewallet\fR
.IP
Do not load the wallet and disable wallet RPC calls
.HP
\fB\-instantsendnotify=\fR<cmd>
.IP
Execute command when a wallet InstantSend transaction is successfully
locked (%s in cmd is replaced by TxID)
.HP
\fB\-keypool=\fR<n>
.IP
Set key pool size to <n> (default: 1000)
.HP
\fB\-rescan=\fR<mode>
.IP
Rescan the block chain for missing wallet transactions on startup (1 =
start from wallet creation time, 2 = start from genesis block)
.HP
\fB\-salvagewallet\fR
.IP
Attempt to recover private keys from a corrupt wallet on startup
.HP
\fB\-spendzeroconfchange\fR
.IP
Spend unconfirmed change when sending transactions (default: 1)
.HP
\fB\-upgradewallet\fR
.IP
Upgrade wallet to latest format on startup
.HP
\fB\-wallet=\fR<path>
.IP
Specify wallet database path. Can be specified multiple times to load
multiple wallets. Path is interpreted relative to <walletdir> if
it is not absolute, and will be created if it does not exist (as
a directory containing a wallet.dat file and log files). For
backwards compatibility this will also accept names of existing
data files in <walletdir>.)
.HP
\fB\-walletbackupsdir=\fR<dir>
.IP
Specify full path to directory for automatic wallet backups (must exist)
.HP
\fB\-walletbroadcast\fR
.IP
Make the wallet broadcast transactions (default: 1)
.HP
\fB\-walletdir=\fR<dir>
.IP
Specify directory to hold wallets (default: <datadir>/wallets if it
exists, otherwise <datadir>)
.HP
\fB\-walletnotify=\fR<cmd>
.IP
Execute command when a wallet transaction changes (%s in cmd is replaced
by TxID)
.HP
\fB\-zapwallettxes=\fR<mode>
.IP
Delete all wallet transactions and only recover those parts of the
blockchain through \fB\-rescan\fR on startup (1 = keep tx meta data e.g.
account owner and payment request information, 2 = drop tx meta
data)
.HP
\fB\-discardfee=\fR<amt>
.IP
The fee rate (in NEOXA/kB) that indicates your tolerance for discarding
change by adding it to the fee (default: 0.0001). Note: An output
is discarded if it is dust at this rate, but we will always
discard up to the dust relay fee and a discard fee above that is
limited by the fee estimate for the longest target
.HP
\fB\-fallbackfee=\fR<amt>
.IP
A fee rate (in DASH/kB) that will be used when fee estimation has
insufficient data (default: 0.00001)
.HP
\fB\-mintxfee=\fR<amt>
.IP
Fees (in NEOXA/kB) smaller than this are considered zero fee for
transaction creation (default: 0.00001)
.HP
\fB\-paytxfee=\fR<amt>
.IP
Fee (in NEOXA/kB) to add to transactions you send (default: 0.00)
.HP
\fB\-txconfirmtarget=\fR<n>
.IP
If paytxfee is not set, include enough fee so transactions begin
confirmation on average within n blocks (default: 6)
.HP
\fB\-hdseed=\fR<hex>
.IP
User defined seed for HD wallet (should be in hex). Only has effect
during wallet creation/first start (default: randomly generated)
.HP
\fB\-mnemonic=\fR<text>
.IP
User defined mnemonic for HD wallet (bip39). Only has effect during
wallet creation/first start (default: randomly generated)
.HP
\fB\-mnemonicpassphrase=\fR<text>
.IP
User defined mnemonic passphrase for HD wallet (BIP39). Only has effect
during wallet creation/first start (default: empty string)
.HP
\fB\-usehd\fR
.IP
Use hierarchical deterministic key generation (HD) after BIP39/BIP44.
Only has effect during wallet creation/first start (default: 0)
.HP
\fB\-keepass\fR
.IP
Use KeePass 2 integration using KeePassHttp plugin (default: 0)
.HP
\fB\-keepassid=\fR<id>
.IP
KeePassHttp id for the established association
.HP
\fB\-keepasskey=\fR<key>
.IP
KeePassHttp key for AES encrypted communication with KeePass
.HP
\fB\-keepassname=\fR<name>
.IP
Name to construct url for KeePass entry that stores the wallet
passphrase
.HP
\fB\-keepassport=\fR<port>
.IP
Connect to KeePassHttp on port <port> (default: 19455)
.HP
\fB\-coinjoinamount=\fR<n>
.IP
Target CoinJoin balance (2\-21000000, default: 1000)
.HP
\fB\-coinjoinautostart\fR
.IP
Start CoinJoin automatically (0\-1, default: 0)
.HP
\fB\-coinjoindenomsgoal=\fR<n>
.IP
Try to create at least N inputs of each denominated amount (10\-100000,
default: 50)
.HP
\fB\-coinjoindenomshardcap=\fR<n>
.IP
Create up to N inputs of each denominated amount (10\-100000, default:
300)
.HP
\fB\-coinjoinmultisession\fR
.IP
Enable multiple CoinJoin mixing sessions per block, experimental (0\-1,
default: 0)
.HP
\fB\-coinjoinrounds=\fR<n>
.IP
Use N separate masternodes for each denominated input to mix funds
(2\-16, default: 4)
.HP
\fB\-coinjoinsessions=\fR<n>
.IP
Use N separate masternodes in parallel to mix funds (1\-10, default: 4)
.HP
\fB\-enablecoinjoin\fR
.IP
Enable use of CoinJoin for funds stored in this wallet (0\-1, default: 0)
.PP
PrivateSend options:
.HP
\fB\-enableprivatesend\fR
.IP
Enable use of PrivateSend for funds stored in this wallet (0\-1, default:
0)
.HP
\fB\-privatesendautostart\fR
.IP
Start PrivateSend automatically (0\-1, default: 0)
.HP
\fB\-privatesendmultisession\fR
.IP
Enable multiple PrivateSend mixing sessions per block, experimental
(0\-1, default: 0)
.HP
\fB\-privatesendsessions=\fR<n>
.IP
Use N separate masternodes in parallel to mix funds (1\-10, default: 4)
.HP
\fB\-privatesendrounds=\fR<n>
.IP
Use N separate masternodes for each denominated input to mix funds
(2\-16, default: 4)
.HP
\fB\-privatesendamount=\fR<n>
.IP
Target PrivateSend balance (2\-21000000, default: 1000)
.HP
\fB\-privatesenddenomsgoal=\fR<n>
.IP
Try to create at least N inputs of each denominated amount (10\-100000,
default: 50)
.HP
\fB\-privatesenddenomshardcap=\fR<n>
.IP
Create up to N inputs of each denominated amount (10\-100000, default:
300)
.PP
ZeroMQ notification options:
.HP
\fB\-zmqpubhashblock=\fR<address>
.IP
Enable publish hash block in <address>
.HP
\fB\-zmqpubhashgovernanceobject=\fR<address>
.IP
Enable publish hash of governance objects (like proposals) in <address>
.HP
\fB\-zmqpubhashgovernancevote=\fR<address>
.IP
Enable publish hash of governance votes in <address>
.HP
\fB\-zmqpubhashinstantsenddoublespend=\fR<address>
.IP
Enable publish transaction hashes of attempted InstantSend double spend
in <address>
.HP
\fB\-zmqpubhashrecoveredsig=\fR<address>
.IP
Enable publish message hash of recovered signatures (recovered by LLMQs)
in <address>
.HP
\fB\-zmqpubhashtx=\fR<address>
.IP
Enable publish hash transaction in <address>
.HP
\fB\-zmqpubhashtxlock=\fR<address>
.IP
Enable publish hash transaction (locked via InstantSend) in <address>
.HP
\fB\-zmqpubrawblock=\fR<address>
.IP
Enable publish raw block in <address>
.HP
\fB\-zmqpubrawinstantsenddoublespend=\fR<address>
.IP
Enable publish raw transactions of attempted InstantSend double spend in
<address>
.HP
\fB\-zmqpubrawrecoveredsig=\fR<address>
.IP
Enable publish raw recovered signatures (recovered by LLMQs) in
<address>
.HP
\fB\-zmqpubrawtx=\fR<address>
.IP
Enable publish raw transaction in <address>
.HP
\fB\-zmqpubrawtxlock=\fR<address>
.IP
Enable publish raw transaction (locked via InstantSend) in <address>
.PP
Debugging/Testing options:
.HP
\fB\-debug=\fR<category>
.IP
Output debugging information (default: 0, supplying <category> is
optional). If <category> is not supplied or if <category> = 1,
output all debugging information. <category> can be: net, tor,
mempool, http, bench, zmq, db, rpc, estimatefee, addrman,
selectcoins, reindex, cmpctblock, rand, prune, proxy, mempoolrej,
libevent, coindb, qt, leveldb, chainlocks, gobject, instantsend,
keepass, llmq, llmq\-dkg, llmq\-sigs, mnpayments, mnsync, coinjoin,
spork, netconn.
.HP
\fB\-debugexclude=\fR<category>
.IP
Exclude debugging information for a category. Can be used in conjunction
with \fB\-debug\fR=\fI\,1\/\fR to output debug logs for all categories except one
or more specified categories.
.HP
\fB\-disablegovernance\fR
.IP
Disable governance validation (0\-1, default: 0)
.HP
\fB\-help\-debug\fR
.IP
Show all debugging options (usage: \fB\-\-help\fR \fB\-help\-debug\fR)
.HP
\fB\-highsubsidyblocks=\fR<n>
.IP
The number of blocks with a higher than normal subsidy to mine at the
start of a devnet (default: 0)
.HP
\fB\-highsubsidyfactor=\fR<n>
.IP
The factor to multiply the normal block subsidy by while in the
highsubsidyblocks window of a devnet (default: 1)
.HP
\fB\-llmqchainlocks=\fR<quorum name>
.IP
Override the default LLMQ type used for ChainLocks on a devnet. Allows
using ChainLocks with smaller LLMQs. (default: llmq_50_60)
.HP
\fB\-llmqdevnetparams=\fR<size:threshold>
.IP
Override the default LLMQ size for the LLMQ_DEVNET quorum (default:
10:6)
.HP
\fB\-llmqinstantsend=\fR<quorum name>
.IP
Override the default LLMQ type used for InstantSend on a devnet. Allows
using InstantSend with smaller LLMQs. (default: llmq_50_60)
.HP
\fB\-llmqtestparams=\fR<size:threshold>
.IP
Override the default LLMQ size for the LLMQ_TEST quorum (default: 3:2)
.HP
\fB\-logips\fR
.IP
Include IP addresses in debug output (default: 0)
.HP
\fB\-logtimestamps\fR
.IP
Prepend debug output with timestamp (default: 1)
.HP
\fB\-maxtxfee=\fR<amt>
.IP
Maximum total fees (in NEOXA) to use in a single wallet transaction or
raw transaction; setting this too low may abort large
transactions (default: 0.10)
.HP
\fB\-minimumdifficultyblocks=\fR<n>
.IP
The number of blocks that can be mined with the minimum difficulty at
the start of a devnet (default: 0)
.HP
\fB\-minsporkkeys=\fR<n>
.IP
Overrides minimum spork signers to change spork value. Only useful for
regtest and devnet. Using this on mainnet or testnet will ban
you.
.HP
\fB\-printtoconsole\fR
.IP
Send trace/debug info to console instead of debug.log file
.HP
\fB\-shrinkdebugfile\fR
.IP
Shrink debug.log file on client startup (default: 1 when no \fB\-debug\fR)
.HP
\fB\-sporkaddr=\fR<neoxaaddress>
.IP
Override spork address. Only useful for regtest and devnet. Using this
on mainnet or testnet will ban you.
.HP
\fB\-sporkkey=\fR<privatekey>
.IP
Set the private key to be used for signing spork messages.
.HP
\fB\-uacomment=\fR<cmt>
.IP
Append comment to the user agent string
.PP
Chain selection options:
.HP
\fB\-devnet=\fR<name>
.IP
Use devnet chain with provided name
.HP
\fB\-testnet\fR
.IP
Use the test chain
.PP
Node relay options:
.HP
\fB\-bytespersigop\fR
.IP
Minimum bytes per sigop in transactions we relay and mine (default: 20)
.HP
\fB\-datacarrier\fR
.IP
Relay and mine data carrier transactions (default: 1)
.HP
\fB\-datacarriersize\fR
.IP
Maximum size of data in data carrier transactions we relay and mine
(default: 83)
.HP
\fB\-minrelaytxfee=\fR<amt>
.IP
Fees (in NEOXA/kB) smaller than this are considered zero fee for
relaying, mining and transaction creation (default: 0.00001)
.HP
\fB\-whitelistforcerelay\fR
.IP
Force relay of transactions from whitelisted peers even if they violate
local relay policy (default: 1)
.HP
\fB\-whitelistrelay\fR
.IP
Accept relayed transactions received from whitelisted peers even when
not relaying transactions (default: 1)
.PP
Block creation options:
.HP
\fB\-blockmaxsize=\fR<n>
.IP
Set maximum block size in bytes (default: 2000000)
.HP
\fB\-blockmintxfee=\fR<amt>
.IP
Set lowest fee rate (in NEOXA/kB) for transactions to be included in
block creation. (default: 0.00001)
.PP
RPC server options:
.HP
\fB\-rest\fR
.IP
Accept public REST requests (default: 0)
.HP
\fB\-rpcallowip=\fR<ip>
.IP
Allow JSON\-RPC connections from specified source. Valid for <ip> are a

doc/release-notes.md
(note: this is a temporary file, to be added-to by anybody, and moved to
release-notes at release time)

Neoxa Core version *version* is now available from:

  <https://neoxa.net/bin/neoxa-core-*version*/>

This is a new major version release, including new features, various bugfixes
and performance improvements, as well as updated translations.

Please report bugs using the issue tracker at GitHub:

  <https://github.com/NeoxaChain/Neoxa/issues>

To receive security and update notifications, please subscribe to:

  <https://neoxa.net/en/list/announcements/join/>

How to Upgrade
==============

If you are running an older version, shut it down. Wait until it has completely
shut down (which might take a few minutes for older versions), then run the 
installer (on Windows) or just copy over `/Applications/Neoxa-Qt` (on Mac)
or `neoxad`/`neoxa-qt` (on Linux).

The first time you run version 0.15.0, your chainstate database will be converted to a
new format, which will take anywhere from a few minutes to half an hour,
depending on the speed of your machine.

Note that the block database format also changed in version 0.8.0 and there is no
automatic upgrade code from before version 0.8 to version 0.15.0. Upgrading
directly from 0.7.x and earlier without redownloading the blockchain is not supported.
However, as usual, old wallet versions are still supported.

Downgrading warning
-------------------

The chainstate database for this release is not compatible with previous
releases, so if you run 0.15 and then decide to switch back to any
older version, you will need to run the old release with the `-reindex-chainstate`
option to rebuild the chainstate data structures in the old format.

If your node has pruning enabled, this will entail re-downloading and
processing the entire blockchain.

Compatibility
==============

Neoxa Core is extensively tested on multiple operating systems using
the Linux kernel, macOS 10.8+, and Windows Vista and later. Windows XP is not supported.

Neoxa Core should also work on most other Unix-like systems but is not
frequently tested on them.

Notable changes
===============

Miner block size limiting deprecated
------------------------------------

Though blockmaxweight has been preferred for limiting the size of blocks returned by
getblocktemplate since 0.13.0, blockmaxsize remained as an option for those who wished
to limit their block size directly. Using this option resulted in a few UI issues as
well as non-optimal fee selection and ever-so-slightly worse performance, and has thus
now been deprecated. Further, the blockmaxsize option is now used only to calculate an
implied blockmaxweight, instead of limiting block size directly. Any miners who wish
to limit their blocks by size, instead of by weight, will have to do so manually by
removing transactions from their block template directly.

HD-wallets by default
---------------------
Due to a backward-incompatible change in the wallet database, wallets created
with version 0.16.0 will be rejected by previous versions. Also, version 0.16.0
will only create hierarchical deterministic (HD) wallets.

Low-level RPC changes
----------------------
- The "currentblocksize" value in getmininginfo has been removed.
- The deprecated RPC `getinfo` was removed. It is recommended that the more specific RPCs are used:
  * `getblockchaininfo`
  * `getnetworkinfo`
  * `getwalletinfo`
  * `getmininginfo`

- `dumpwallet` no longer allows overwriting files. This is a security measure
  as well as prevents dangerous user mistakes.

Credits
=======

Thanks to everyone who directly contributed to this release:


As well as everyone that helped translating on [Transifex](https://www.transifex.com/projects/p/neoxa/).


src/neoxad.cpp
// Copyright (c) 2009-2010 Satoshi Nakamoto
// Copyright (c) 2009-2015 The Bitcoin Core developers
// Copyright (c) 2014-2019 The Dash Core developers
// Copyright (c) 2020 The Neoxa developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

#if defined(HAVE_CONFIG_H)
#include "config/neoxa-config.h"
#endif

#include "chainparams.h"
#include "clientversion.h"
#include "compat.h"
#include "fs.h"
#include "rpc/server.h"
#include "init.h"
#include "noui.h"
#include "scheduler.h"
#include "util.h"
#include "httpserver.h"
#include "httprpc.h"
#include "utilstrencodings.h"
#include "stacktraces.h"

#include <boost/thread.hpp>

#include <stdio.h>

/* Introduction text for doxygen: */

/*! \mainpage Developer documentation
 *
 * \section intro_sec Introduction
 *
 * This is the developer documentation of the reference client for an experimental new digital currency called Neoxa (https://www.neoxa.net/),
 * which enables instant payments to anyone, anywhere in the world. Neoxa uses peer-to-peer technology to operate
 * with no central authority: managing transactions and issuing money are carried out collectively by the network.
 *
 * The software is a community-driven open source project, released under the MIT license.
 *
 * \section Navigation
 * Use the buttons <code>Namespaces</code>, <code>Classes</code> or <code>Files</code> at the top of the page to start navigating the code.
 */

void WaitForShutdown(boost::thread_group* threadGroup)
{
    bool fShutdown = ShutdownRequested();
    // Tell the main threads to shutdown.
    while (!fShutdown)
    {
        MilliSleep(200);
        fShutdown = ShutdownRequested();
    }
    if (threadGroup)
    {
        Interrupt(*threadGroup);
        threadGroup->join_all();
    }
}

//////////////////////////////////////////////////////////////////////////////
//
// Start
//
bool AppInit(int argc, char* argv[])
{
    boost::thread_group threadGroup;
    CScheduler scheduler;

    bool fRet = false;

    //
    // Parameters
    //
    // If Qt is used, parameters/neoxa.conf are parsed in qt/neoxa.cpp's main()
    gArgs.ParseParameters(argc, argv);

    if (gArgs.IsArgSet("-printcrashinfo")) {
        std::cout << GetCrashInfoStrFromSerializedStr(gArgs.GetArg("-printcrashinfo", "")) << std::endl;
        return true;
    }

    // Process help and version before taking care about datadir
    if (gArgs.IsArgSet("-?") || gArgs.IsArgSet("-h") ||  gArgs.IsArgSet("-help") || gArgs.IsArgSet("-version"))
    {
        std::string strUsage = strprintf(_("%s Daemon"), _(PACKAGE_NAME)) + " " + _("version") + " " + FormatFullVersion() + "\n";

        if (gArgs.IsArgSet("-version"))
        {
            strUsage += FormatParagraph(LicenseInfo());
        }
        else
        {
            strUsage += "\n" + _("Usage:") + "\n" +
                  "  neoxad [options]                     " + strprintf(_("Start %s Daemon"), _(PACKAGE_NAME)) + "\n";

            strUsage += "\n" + HelpMessage(HMM_BITCOIND);
        }

        fprintf(stdout, "%s", strUsage.c_str());
        return true;
    }

    try
    {
        bool datadirFromCmdLine = gArgs.IsArgSet("-datadir");
        if (datadirFromCmdLine && !fs::is_directory(GetDataDir(false)))
        {
            fprintf(stderr, "Error: Specified data directory \"%s\" does not exist.\n", gArgs.GetArg("-datadir", "").c_str());
            return false;
        }
        try
        {
            gArgs.ReadConfigFile(gArgs.GetArg("-conf", BITCOIN_CONF_FILENAME));
        } catch (const std::exception& e) {
            fprintf(stderr,"Error reading configuration file: %s\n", e.what());
            return false;
        }
        if (!datadirFromCmdLine && !fs::is_directory(GetDataDir(false)))
        {
            fprintf(stderr, "Error: Specified data directory \"%s\" from config file does not exist.\n", gArgs.GetArg("-datadir", "").c_str());
            return EXIT_FAILURE;
        }
        // Check for -testnet or -regtest parameter (Params() calls are only valid after this clause)
        try {
            SelectParams(ChainNameFromCommandLine());
        } catch (const std::exception& e) {
            fprintf(stderr, "Error: %s\n", e.what());
            return false;
        }

        // Error out when loose non-argument tokens are encountered on command line
        for (int i = 1; i < argc; i++) {
            if (!IsSwitchChar(argv[i][0])) {
                fprintf(stderr, "Error: Command line contains unexpected token '%s', see neoxad -h for a list of options.\n", argv[i]);
                exit(EXIT_FAILURE);
            }
        }

        // -server defaults to true for bitcoind but not for the GUI so do this here
        gArgs.SoftSetBoolArg("-server", true);
        // Set this early so that parameter interactions go to console
        InitLogging();
        InitParameterInteraction();
        if (!AppInitBasicSetup())
        {
            // InitError will have been called with detailed error, which ends up on console
            exit(EXIT_FAILURE);
        }
        if (!AppInitParameterInteraction())
        {
            // InitError will have been called with detailed error, which ends up on console
            exit(EXIT_FAILURE);
        }
        if (!AppInitSanityChecks())
        {
            // InitError will have been called with detailed error, which ends up on console
            exit(EXIT_FAILURE);
        }
        if (gArgs.GetBoolArg("-daemon", false))
        {
#if HAVE_DECL_DAEMON
            fprintf(stdout, "Neoxa Core server starting\n");

            // Daemonize
            if (daemon(1, 0)) { // don't chdir (1), do close FDs (0)
                fprintf(stderr, "Error: daemon() failed: %s\n", strerror(errno));
                return false;
            }
#else
            fprintf(stderr, "Error: -daemon is not supported on this operating system\n");
            return false;
#endif // HAVE_DECL_DAEMON
        }
        // Lock data directory after daemonization
        if (!AppInitLockDataDirectory())
        {
            // If locking the data directory failed, exit immediately
            exit(EXIT_FAILURE);
        }
        fRet = AppInitMain(threadGroup, scheduler);
    } catch (...) {
        PrintExceptionContinue(std::current_exception(), "AppInit()");
    }

    if (!fRet)
    {
        Interrupt(threadGroup);
        threadGroup.join_all();
    } else {
        WaitForShutdown(&threadGroup);
    }
    Shutdown();

    return fRet;
}

int main(int argc, char* argv[])
{
    RegisterPrettyTerminateHander();
    RegisterPrettySignalHandlers();

    SetupEnvironment();

    // Connect neoxad signal handlers
    noui_connect();

    return (AppInit(argc, argv) ? EXIT_SUCCESS : EXIT_FAILURE);
}
